syncosaurus/syncosaurus.js

```diff
@@ -54,13 +57,15 @@ export default class Syncosaurus {
         //re-run the query
 
         if (
-          subscription.keys[notificationKey] &&
+          (subscription.keys[notificationKey] || subscription.scanFlag) &&
           !executedSubscriptions[subIdx]
         ) {
           let newSubKeys = {};
+          let scan = {};
```

Comment:
Wondering if there is a reason why `newSubKeys` and `scan` are newly initialized then immediately followed by an instantiation of a new transaction? Is there a reason against initializing a new object for `newSubKeys` and `scan`, and then being able to access them via a prooperty of a transaction instance?

Reply:
That's a good point. I guess they could just be properties of the transaction class and then we could access them directly as a property of the transaction class.

Reply:
I implemented the `newSubKeys` and `scan` objects directly on the transaction object as suggested.